### DISCUSSION TITLE
Name and Image is always null in OneLogin in v4

### DISCUSSION THREAD

**Discussion by heylols** [nextauthjs/next-auth#3182] [Help]
### Question 💬

I have the OneLogin setup. And am using MongoDB as database. Here's the relevant portion of code from the [...nextauth].js file:

```
        callbacks: {
            async session({ session, token, user }) {
                
                //-- pass the ID of the user from the database
                session.user.uid = user.id
                
                return session
            }
        },

        adapter: MongoDBAdapter({
            db: (await mongodb).db("abc_coll")
        }),
```

The problem is, the `users` table contains the `name` and `image` field always as null. When I enabled the debug in NextAuth, I can see something similar to this:

```
[next-auth][debug][OAUTH_CALLBACK_RESPONSE] {
   ...... 

  OAuthProfile: {
    sub: 'xxxxxxxx',
    email: 'xxx@xxxxx.com',
    preferred_username: 'xxx@xxxxxxx.com',
    name: 'ABC Cde',
    updated_at: 1636654238,
    given_name: 'ABC',
    family_name: 'Cde',
    at_hash: 'aadsasdadads',
    sid: 'adasdasdasd',
    aud: 'asdasdadasd',
    exp: 1636662905,
    iat: 1636655705,
    iss: 'https://mywebsite.com/oidc/2',
    user: null
  }
}
```

Kindly note that I have masked some of the values. As you can see that the OAuthProfile has the `name`. How can I access this data and use it?

### How to reproduce ☕️

Unable to post the whole code.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
Check out the `profile` callback: https://next-auth.js.org/adapters/models#user

Here is the data being saved if you create the user with OneLogin:

https://github.com/nextauthjs/next-auth/blob/e9ac11b4b25d6d14e76a6d2f0fe37144c59902f6/src/providers/onelogin.js#L12-L15

return any additional fields if you want to, and those will be added on user creation

**Reply by heylols** [reply]
Thank you! 

I added the following :
```
                profile(profile) {
                    return {
                        id: profile.sub,
                        name: profile.name,
                        email: profile.email,
                        image: profile.picture,
                        username: profile.preferred_username,
                    }
                },
```

That's seem to be working. But the picture field am not seeing in the OAuthProfile. Hence it's always null, even though I set an image through the OneLogin dashboard. Any thoughts?
